code/model/modelread.cpp

```diff
@@ -2373,8 +2373,9 @@ modelread_status read_model_file_no_subsys(polymodel * pm, const char* filename,
 						for (j = 0; j < bay->num_slots; j++) {
 							cfread_vector( &(bay->pnt[j]), fp );
 							cfread_vector( &(bay->norm[j]), fp );
-							if(vm_vec_mag(&(bay->norm[j])) <= 0.0f) {
-								Warning(LOCATION, "Model '%s' dock point '%s' has a null normal. ", filename, bay->name);
+							if(!vm_vec_is_normalized(&(bay->norm[j]))) {
+								Warning(LOCATION, "Model '%s' dock point '%s' normal %d is not normalized!", filename, bay->name, j);
+								vm_vec_rand_vec(&(bay->norm[j]));
```

Comment:
This should not, imo, even be a warning, if it isn't null, then it can and should simply be normalized right here without notification.

Also `vm_vec_rand_vec`?? I assume it wasn't intentional to randomize this?

Reply:
Hmm, I was trying to handle the null case and the non-normalized case using the same code, but those are good points.  I've pushed a modification.

Reply:
I'm still a little confused about the random vector. If we are forced to generate something, it should simply be like `vmd_z_vector` or something, not random. If we're going to give them something probably wrong (if at least valid), we should at least be consistent about it, being inconsistent can easily cause even more problems.

Reply:
Mm, fair enough.